src/scope/models/version.ts

```diff
@@ -52,6 +55,122 @@ type ExternalHead = { head: Ref; laneId: LaneId };
 type SquashData = { previousParents: Ref[]; laneId: LaneId };
 type VersionOrigin = { id: { scope: string; name: string }; lane?: { scope: string; name: string; hash: string } };
 
+export type PackagesMap = Map<string, PackageAttributes>;
+
+export type PackageAttributes = PackageInfo & {
+  component?: {
+    scope: string;
+    name: string;
+  };
+};
+
+export type DependencyEdge = {
+  id: string;
+  neighbours: DependencyNeighbour[];
+  attr?: {
+    pkgId?: string;
+    transitivePeerDependencies?: string[];
+  };
+};
+
+export type DependencyNeighbour = {
+  id: string;
+  /**
+   * This is true when the dependency is from optionalDependencies.
+   */
+  optional?: boolean;
+  name?: string;
+  specifier?: string;
+  lifecycle?: 'runtime' | 'dev';
+};
+
+const DEPENDENCIES_GRAPH_SCHEMA_VERSION = '1.0';
+
+export class DependenciesGraph {
+  static ROOT_EDGE_ID = '.';
+
+  schemaVersion: string;
+  packages: PackagesMap;
+  edges: DependencyEdge[];
+
+  constructor({
+    packages,
+    edges,
+    schemaVersion,
+  }: {
+    packages: PackagesMap;
+    edges: DependencyEdge[];
+    schemaVersion?: string;
+  }) {
+    this.packages = packages;
+    this.edges = edges;
+    this.schemaVersion = schemaVersion ?? DEPENDENCIES_GRAPH_SCHEMA_VERSION;
+  }
+
+  serialize(): string {
+    return JSON.stringify({
+      schemaVersion: this.schemaVersion,
+      packages: Object.fromEntries(this.packages.entries()),
+      edges: this.edges,
+    });
+  }
+
+  static deserialize(data: string): DependenciesGraph | undefined {
+    const parsed = JSON.parse(data);
+    // If the schema version is not supported, then we just ignore the data
```

Comment:
I believe that in most cases we will be able to support older data, maybe it's for future, but maybe add a todo here to have a better mechanism , like minimal supported schema version

Reply:
But we don't know what the future version of the data will look like. I understand that maybe when the new version will come out, the new bit CLI will be able to process the previous version too. But the old version of bit CLI can't process the future format.